diesel/src/types/impls/date_and_time/chrono.rs

```diff
@@ -0,0 +1,127 @@
+//! This module makes it possible to map `chrono::DateTime` values to postgres `Date`
+//! and `Timestamp` fields. It is enabled with the `chrono` feature.
+
+extern crate chrono;
+extern crate byteorder as localbyteorder; // conflicts otherwise
+
+use std::fmt::Display;
+use std::error::Error;
+use std::io::Write;
+use self::localbyteorder::{ReadBytesExt, WriteBytesExt, BigEndian};
+use self::chrono::{Duration, NaiveDate, NaiveDateTime, DateTime, UTC, Local, FixedOffset, TimeZone};
+
+use expression::{Expression, NonAggregate};
+use query_builder::{QueryBuilder, BuildQueryResult};
+use types::{FromSql, IsNull, Timestamp, ToSql};
+
+
+// Postgres timestamps start from January 1st 2000.
+fn base() -> NaiveDateTime {
+    NaiveDate::from_ymd(2000, 1, 1).and_hms(0, 0, 0)
+}
+
+impl<Tz: TimeZone> Expression for DateTime<Tz>
+    where Tz::Offset: Display {
+    type SqlType = Timestamp;
+
+    fn to_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
+        let formatted_string = format!("{}", &self.format("%Y-%m-%d %H:%M:%S %:z"));
+        out.push_sql(&formatted_string);
+        Ok(())
+    }
+}
```

Comment:
These definitely shouldn't implement `Expression`, they can be used as bind params simply by implementing `ToSql`